### DISCUSSION TITLE
MIDI.read

### DISCUSSION THREAD

**Discussion by oxxe9** [lathoub/Arduino-AppleMIDI-Library#158] [General]
Hello! 

For  a project i need the value of the note and velocity. Channel would also be nice, but this two are first of all enough. 
When i read out MIDI.read like this:

Serial.println(MIDI.read());
I only get back NoteOn, or NoteOff connected with the number of the note. 

For example: 
16:46:15.851 -> NoteOn 60
16:46:15.955 -> NoteOff 60

Is there any way to read the other values? 

Many, many thanks in advance! 

Regards,

oxxe9

**Comment by lathoub** [comment]
Can you post a more complete minimum code set.
Also `Serial.println(MIDI.read());` will only print a bool indicating a message was received.

Did you have a look at the [examples](https://github.com/lathoub/Arduino-AppleMIDI-Library/blob/master/examples/AVR_NoteOnOffEverySec/AVR_NoteOnOffEverySec.ino)?

https://github.com/lathoub/Arduino-AppleMIDI-Library/blob/d8a4e987f7a714a619001cbb4584dcefe756379e/examples/AVR_NoteOnOffEverySec/AVR_NoteOnOffEverySec.ino#L48-L50
